### ISSUE TITLE
Expandable potion system

### ISSUE THREAD

**Issue by Grumblin** [Adriftus-Studios/network-script-data#79]
For survival, we need a up-gradable potion system; the system needs to be 'easily' added to, and expandable (allowing more potions to be added into the mix later down the line)

To start off, I need an upgrade able instant health potion (only the 2nd tier matters) - it needs to do the following:

- [ ] Upgrade
  - [ ] putting 2 of the tier below it into a crafting grid, and then a secondary ingredient will upgrade the potion to 1 of the next tier
  - [ ] each potion has a limited amount of uses based on the tier (listed below) and then be destroyed
  - [ ] each potion will heal a certain amount of health (listed below)
  - [ ] cannot upgrade potions that have less than the original use count - can't use them in a recipe for a new one either

- [ ] **Tier 1** - vanilla potion
- [ ] **Tier 2**
  - [ ] brown mushroom secondary
  - [ ] heals 10 hearts on consume
  - [ ] has 3 uses
- [ ] **Tier 3**
  - [ ] glowstone secondary
  - [ ] heals 15 hearts on consume
  - [ ] has 5 uses
- [ ] **Tier 4**
  - [ ] fire charge secondary
  - [ ] heals 20 hearts on consume
  - [ ] has 8 uses
- [ ] **Tier 5**
  - [ ] block of redstone secondary
  - [ ] heals 25 hearts
  - [ ] has 12 uses
- [ ] **Tier 6**
  - [ ] ancient debris secondary
  - [ ] heals 30 hearts
  - [ ] has 18 uses
- [ ] **Tier 7**
  - [ ] nether star secondary
  - [ ] heals 45 hearts
  - [ ] has 25 uses
- [ ] **Tier 8**
  - [ ] block of diamond secondary
  - [ ] heals 70 hearts
  - [ ] has 35 uses
- [ ] **Tier 9**
  - [ ] netherite ingot secondary
  - [ ] heals 90 hearts
  - [ ] has 40 uses
- [ ] **Tier 10**
  - [ ] Glistening melon secondary
  - [ ] heals max health 
  - [ ] has 50 uses

**Comment by ChrispyMC** [comment]
[Here is a breakdown of the data you have presented.](https://www.youtube.com/watch?v=UztXN2rKQNc)
---
**Number of potion uses**
Let *x* represent the tier of potion.
Let *y* represent the number of uses.
| x | y | 1st diff. |
|:-:|:--:|:--------:|
| 1 (V) | 1 | -- |
| 2 | 3 | 2 |
| 3 | 5 | 2 |
| 4 | 8 | 3 |
| 5 | 12 | 4 |
| 6 | 18 | 6 |
| 7 | 25 | 7 |
| 8 | 35 | 10 |
| 9 | 40 | 5 |
| 10 | 50 | 10 |
---

Knowing that the first differences are not consistent, we cannot use a linear equation to determine a potion's uses. This means that we must define each potion tier's number of uses in a data script.

---
**Amount of hearts healed**
Let *x* represent the tier of potion.
Let *y* represent the amount of hearts healed.
| x | y | 1st diff. |
|:-:|:--:|:--------:|
| 1 (V) | 2 or 4 | -- |
| 2 | 10 | 8 or 6 |
| 3 | 15 | 5 |
| 4 | 20 | 5 |
| 5 | 25 | 5 |
| 6 | 30 | 5 |
| 7 | 45 | 15 |
| 8 | 70 | 25 |
| 9 | 90 | 20 |
| 10 | Max | -- |
---

The first differences here were getting consistent, and there was the equation `Hearts healed (y) = Tier # (x) x 5`. Tier 7 was where its first difference differed from the previous first differences. We can conclude that the amount of hearts healed by one potion use has to be defined in the previously mentioned data script.

**TL;DR: The additional item required in the item recipe, the number of uses, and the number of hearts healed all have to be defined in a data script.**
